src/Domain/Common/Interfaces/IDomainEventEmitter.cs

```diff
@@ -0,0 +1,14 @@
+﻿using SSW.CleanArchitecture.Domain.Common.Base;
+
+namespace SSW.CleanArchitecture.Domain.Common.Interfaces;
+
+public interface IDomainEventEmitter
```

Comment:
Personally, I'm not a big fan of this name.  This interface is not just for 'emitting events'.  It's also for raising them.